### PR TITLE
samples: Bluetooth: Broadcast audio sink USB headset on nRF52 Series

### PR DESCRIPTION
**Pull request by cvinayak**
Broadcast audio sink USB headset on nRF52 Series with USB support.

### CODE REVIEW

File: samples/bluetooth/broadcast_audio_sink/boards/nrf52833dk_nrf52833.overlay [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+		compatible = "usb-audio-hs";
+		mic-feature-mute;
+		mic-channel-l;
+		mic-channel-r;
+
+		hp-feature-mute;
+		hp-channel-l;
+		hp-channel-r;
+	};
+};
```
Comment (Thalley) [inline samples/bluetooth/broadcast_audio_sink/boards/nrf52833dk_nrf52833.overlay:15]: Is there a guideline for when these should be added to the overlay files, and when they should be added to the board definitions? 

e.g. the 5340 ADK has hs_0 in the board definition.

File: samples/bluetooth/broadcast_audio_sink/boards/nrf52833dk_nrf52833.conf [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# The LC3 codec uses a large amount of stack. This app runs the codec in the work-queue, hence
+# increase stack size for that thread.
+CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE=4096
```
Comment (Thalley) [inline samples/bluetooth/broadcast_audio_sink/boards/nrf52833dk_nrf52833.conf:3]: We may want to just set this as the default stack size for the entire sample

File: samples/bluetooth/broadcast_audio_sink/boards/nrf5340_audio_dk_nrf5340_cpuapp.conf [hunk 3]
```diff
@@ -1,7 +1,5 @@
 # The LC3 codec uses a large amount of stack. This app runs the codec in the work-queue, hence
-# inctease stack size for that thread.
+# increase stack size for that thread.
 CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE=4096
```
Comment (Thalley) [inline samples/bluetooth/broadcast_audio_sink/boards/nrf5340_audio_dk_nrf5340_cpuapp.conf:3]: Hmm, the comment here is actually no longer true. The LC3 decoder runs in its own thread in the sample now. Perhaps the increase to `CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE` should just be removed

File: samples/bluetooth/broadcast_audio_sink/src/main.c [hunk 4]
```diff
@@ -1044,8 +1044,8 @@ static int init(void)
 
 	usb_audio_register(hs_dev, &usb_ops);
 	err = usb_enable(NULL);
-	if (err != 0) {
-		printk("Failed to enable USB\n");
+	if (err && err != -EALREADY) {
+		printk("Failed to enable USB (%d)", err);
```
Comment (Thalley) [inline samples/bluetooth/broadcast_audio_sink/src/main.c:1048]: When would it return -EALREADY? 

If something else enables USB?

File: samples/bluetooth/broadcast_audio_source/boards/nrf5340_audio_dk_nrf5340_cpuapp.conf [hunk 5]
```diff
@@ -1,3 +1,5 @@
+# The LC3 codec uses a large amount of stack. This app runs the codec in the work-queue, hence
+# increase stack size for that thread.
 CONFIG_MAIN_STACK_SIZE=4096
```
Comment (Thalley) [inline samples/bluetooth/broadcast_audio_source/boards/nrf5340_audio_dk_nrf5340_cpuapp.conf:3]: This comment does apply to the main stack

Comment (Thalley) [reply samples/bluetooth/broadcast_audio_source/boards/nrf5340_audio_dk_nrf5340_cpuapp.conf:3]: Ditto for a few other places

File: samples/bluetooth/broadcast_audio_source/src/main.c [hunk 6]
```diff
@@ -469,7 +469,7 @@ int main(void)
 	err = usb_enable(NULL);
 	if (err && err != -EALREADY) {
 		printk("Failed to enable USB (%d)", err);
-		return 0;
+		return err;
```
Comment (Thalley) [inline samples/bluetooth/broadcast_audio_source/src/main.c:472]: As per https://docs.zephyrproject.org/latest/develop/languages/cpp/index.html#enabling-c-support, the `main` function should just return `0`:

> Zephyr ignores the return value from main, so applications should not return status information and should, instead, return zero.

**Review by Thalley — CHANGES_REQUESTED** [review]

File: samples/bluetooth/broadcast_audio_sink/src/main.c (hunk 4 above)
Comment (cvinayak) [reply samples/bluetooth/broadcast_audio_sink/src/main.c:1048]: On the USB Dongle Console using USB-CDC and hence is already enabled at boot.

File: samples/bluetooth/broadcast_audio_sink/boards/nrf5340_audio_dk_nrf5340_cpuapp.conf (hunk 3 above)
Comment (cvinayak) [reply samples/bluetooth/broadcast_audio_sink/boards/nrf5340_audio_dk_nrf5340_cpuapp.conf]: Removed... the sample seems to build and work without this increase.

File: samples/bluetooth/broadcast_audio_source/boards/nrf5340_audio_dk_nrf5340_cpuapp.conf (hunk 5 above)
Comment (cvinayak) [reply samples/bluetooth/broadcast_audio_source/boards/nrf5340_audio_dk_nrf5340_cpuapp.conf]: Removed... the sample seems to build and work without this increase.

File: samples/bluetooth/broadcast_audio_source/src/main.c (hunk 6 above)
Comment (cvinayak) [reply samples/bluetooth/broadcast_audio_source/src/main.c]: Reverted

**Review by Thalley — APPROVED** [review]

File: samples/bluetooth/broadcast_audio_sink/boards/nrf52833dk_nrf52833.overlay (hunk 1 above)
Comment (cvinayak) [reply samples/bluetooth/broadcast_audio_sink/boards/nrf52833dk_nrf52833.overlay:15]: These normal DKs with USB support in SoC do not have to have USB Audio enabled, consuming code space for the USB stack. The features being software defined will depend on the applications, hence best defined in the applications boards overlay.

**Review by carlescufi — APPROVED** [review]

**Comment by Thalley** [comment]
@carlescufi I think this may have been prematurely merged: It was merged with a single day, and only approved by Nordic :)

**Comment by larsgk** [comment]
> @carlescufi I think this may have been prematurely merged: It was merged with a single day, and only approved by Nordic :)

It looks like it's mainly adding nrf52x support - with a few minor other changes.  We can do some testing on it to verify, but unless there are strong objections, I think it can stay ;)

**Pull request merged**